src/cr_kyoushi/generator/template.py

```diff
@@ -52,9 +52,165 @@
     )
 
 
+def _normalize_propabilities(
+    propabilities: Sequence[float],
+    check_positive: bool = True,
+) -> List[float]:
+    """Normalizes a propability distribution to sum up to 1.
+
+    Has a floating point error tolerance of up to 1e-8 (same as numpy.choice).
+    If the resulting difference is greater than 1e-8 it is added to the first
+    non zero propability.
+
+    Args:
+        propabilities: The distribution to normalize
+        check_positive: Switch to disable the positive number test.
+                        This makes the function slightly faster
+
+    Raises:
+        Exception: If the distribution sums to 0 or
+                   If one of the propability values is negative
+
+    Returns:
+        The normalized probability distribution.
+    """
+    # only check when requested
+    if check_positive and any(p < 0 for p in propabilities):
+        raise Exception(
+            f"Propabilities must be positive numbers, but got {propabilities}"
+        )
+
+    total = sum(propabilities)
+
+    if abs(1.0 - total) < 1e-8:
+        # if we already have a normalized
+        # distribution then we can just return it
+        return propabilities
```

Comment:
replace this line with `return list(propabilities)` to satisfy mypi

Reply:
Also I recommend setting up pre-commit for this repo so we can avoid semi broken commits